dash/include/dash/Matrix.h

```diff
@@ -135,13 +135,16 @@ template<
   class    PatternT       = TilePattern<NumDimensions, ROW_MAJOR, IndexT> >
 class Matrix
 {
-#if !defined(_CRAYC) && !defined(__INTEL_COMPILER)
```

Comment:
Intel compilers (tested 16.0 and 17.0) don't allow `std::is_trivially_copyable` either. Why did you remove this check?